azavea/raster-foundry
# Feature/lk/operation node body

## Overview
Add operation node body
Fix/Finish canvas interaction


### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
![image](https://user-images.githubusercontent.com/4392704/28126224-e13c2d82-66f6-11e7-96b3-521e2a174c3e.png)
![image](https://user-images.githubusercontent.com/4392704/28126237-eae25f1e-66f6-11e7-8dc6-bf648c5f5357.png)
![image](https://user-images.githubusercontent.com/4392704/28126246-f123d326-66f6-11e7-81d8-76f0f5bf01d0.png)

## Testing Instructions

 * Verify that interaction with the canvas works as you'd expect when zooming, panning, and resizing the window
* Verify that you can no longer move nodes / links in run 
* Verify that the function nodes have the correct inputs listed inside them if there are 2 inputs
* Verify that the zoom levels update
* Verify that clicking apply changes then previewing a node works
* Verify that the graph automatically zooms and centers on load
* Verify that the zoom to fit button works
* Verify that position overrides work by editing the metadata of a node to include something like `positionOverride: {x: 1000, y: 1000}`
I used this function to test it:
```javascript
usePlaceholderDef() {
    this.toolDefinition = {
        id: 'fe1763d4-9b4d-4eba-9968-f7d707869d7f',
        args: [{
            id: '67013aad-21c5-4e7f-b8bd-5910e92aa425',
            type: 'src',
            metadata: {
                label: 'A'
            }
        }, {
            id: 'a3c93582-a086-44e8-8bdf-0b2ab69f91d0',
            type: 'src',
            metadata: {
                label: 'B'
            }
        }],
        apply: '-',
        metadata: {
            label: 'Subtraction (A-B)?',
            positionOverride: {x: 1000, y: 1000}
        }
    };
}
```

Closes #2159 
Closes #2161
Closes #2160 
Closes #2243 
Closes #2242
Closes #2171
